File: hidden-api/src/main/java/android/app/IActivityTaskManager.java
package android.app;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.os.IBinder;
import android.os.IInterface;

import androidx.annotation.DeprecatedSinceApi;
import androidx.annotation.RequiresApi;

/**
 * Compile-time stub for the @hide IActivityTaskManager Binder interface.
 * <p>
 * Hidden-api is compileOnly; at runtime this JVM name resolves to framework's
 * real android.app.IActivityTaskManager.
 */
public interface IActivityTaskManager extends IInterface {

    @RequiresApi(Build.VERSION_CODES.R)
    int startActivity(
            IApplicationThread caller,
            String callingPackage,
            String callingFeatureId,
            Intent intent,
            String resolvedType,
            IBinder resultTo,
            String resultWho,
            int requestCode,
            int flags,
            ProfilerInfo profilerInfo,
            Bundle options
    );

    // 10 parameters version only work at API 29
    @RequiresApi(Build.VERSION_CODES.Q)
    @DeprecatedSinceApi(
            api = Build.VERSION_CODES.R,
            message = "10 parameters version only work at API 29")
    int startActivity(
            IApplicationThread caller,
            String callingPackage,
            Intent intent,
            String resolvedType,
            IBinder resultTo,
            String resultWho,
            int requestCode,
            int flags,
            ProfilerInfo profilerInfo,
            Bundle options
    );
}

File: hidden-api-contract/src/contracts/java/com/xaxaxax/moonclicker/hiddenapi/contracts/IActivityTaskManager.kt
package com.xaxaxax.moonclicker.hiddenapi.contracts

import android.os.Build

/**
 * `callingFeatureId` was inserted in API 30; API 29 has the shorter overload.
 */
internal val I_ACTIVITY_TASK_MANAGER = StubContracts(
    stubs = mapOf("android.app.IActivityTaskManager" to "android.app.IActivityTaskManager"),
    contracts = listOf(
        MemberContract(
            owner = "android.app.IActivityTaskManager",
            member = MethodMember(
                name = "startActivity",
                parameters = listOf(
                    "android.app.IApplicationThread",
                    "java.lang.String",
                    "java.lang.String",
                    "android.content.Intent",
                    "java.lang.String",
                    "android.os.IBinder",
                    "java.lang.String",
                    "int",
                    "int",
                    "android.app.ProfilerInfo",
                    "android.os.Bundle",
                ),
                returns = "int",
            ),
            sinceApi = Build.VERSION_CODES.R,
        ),
        MemberContract(
            owner = "android.app.IActivityTaskManager",
            member = MethodMember(
                name = "startActivity",
                parameters = listOf(
                    "android.app.IApplicationThread",
                    "java.lang.String",
                    "android.content.Intent",
                    "java.lang.String",
                    "android.os.IBinder",
                    "java.lang.String",
                    "int",
                    "int",
                    "android.app.ProfilerInfo",
                    "android.os.Bundle",
                ),
                returns = "int",
            ),
            sinceApi = Build.VERSION_CODES.Q,
            untilApi = Build.VERSION_CODES.Q,
        ),
    ),
)
